crates/snapshot/src/snapshotter.rs

```diff
@@ -0,0 +1,172 @@
+//! Support for snapshotting.
+
+use crate::SnapshotterError;
+use reth_db::database::Database;
+use reth_primitives::{BlockNumber, ChainSpec, TxNumber};
+use reth_provider::{BlockReader, ProviderFactory};
+use std::sync::Arc;
+
+/// Result of [Snapshotter::run] execution.
+pub type SnapshotterResult = Result<(), SnapshotterError>;
```

Comment:
Maybe we return the latest block number and txnumber?

Reply:
which will get consumed by some other hook/process with DbWriteAccess, and will somehow create the pruning jobs

Reply:
good idea, returning `SnapshotTargets` now